moon/controller_round3.py

```diff
@@ -379,35 +355,21 @@ def on_artf(self, timestamp, data):
                         self.publish("desired_movement", [GO_STRAIGHT, 0, SPEED_ON])
 
                         
-                elif math.isinf(self.homebase_final_approach_distance) and self.currently_following_object['object_type'] == 'homebase':
-                    if bbox_size > 200:
-                        if center_x >= 300 and center_x <= 340:
-                            # object reached visually, keep moving forward
-                            self.publish("desired_movement", [GO_STRAIGHT, 0, SPEED_ON])
-                            print(self.time, "app: homebase final frame x=%d y=%d w=%d h=%d" % (data[1], data[2], data[3], data[4]))
-                            self.homebase_final_approach_distance = 30.0
-                            self.on_driving_control(timestamp, "homebase-final")
-                            
-                        elif center_x < 300: # close but wrong angle, turn in place left
-                            self.publish("desired_movement", [0, 0, SPEED_ON])
-                        elif center_x > 340: # close but wrong angle, turn in place right
-                            self.publish("desired_movement", [0, 0, -SPEED_ON])
-                    else:
-                        if center_x < 300: # if homebase to the left, steer left
-                            self.publish("desired_movement", [TURN_ON, 0, SPEED_ON])
-                        elif center_x > 340:
-                            self.publish("desired_movement", [-TURN_ON, 0, SPEED_ON])
-                        else: # if within angle but object too small, keep going straight
-                            self.publish("desired_movement", [GO_STRAIGHT, 0, SPEED_ON])
+                elif self.currently_following_object['object_type'] == 'homebase':
+                    if center_x < 300: # if homebase to the left, steer left
```

Comment:
300 is pixel coordinate for width=640?

Reply:
Yes, the view is 640 pixels wide, 320 is midpoint, so it means turn left or right if the center of the object is more than 20 pixels away from the vertical center line. I suppose we can start introducing constants and parameters once we know what the algorithms and methods are actually going to be.